### PR TITLE
Cleaned up DateFormatter. 

### PR DESCRIPTION
**Pull request by InfinityDeltaX**
Refactored some code in DateFormatter. Added support for changing 'Jul 5' to '5 Jul' when appropriate. Added unit tests for DateFormatter. Moved pull request to develop.

### CODE REVIEW

File: QKSMS/src/main/java/com/moez/QKSMS/common/utils/DateFormatter.java [hunk 1]
```diff
@@ -11,65 +11,90 @@
...
+        boolean shouldBeFlipped = !(Locale.getDefault().equals(Locale.US) || Locale.getDefault().equals(Locale.CANADA));
+
+        if (shouldBeFlipped) {
+            return new SimpleDateFormat(input.toPattern().replaceAll("MMM", "temp").replaceAll("d", "MMM").replaceAll("temp", "d"));
+        } else return input;
+    }
+
     public static String getMessageTimestamp(Context context, long date) {
         SharedPreferences prefs = MainActivity.getPrefs(context);
         boolean isUsing24HourTime = prefs.getBoolean(SettingsFragment.TIMESTAMPS_24H, false);
```
Comment (moezbhatti) [inline QKSMS/src/main/java/com/moez/QKSMS/common/utils/DateFormatter.java]: Unused variable

File: QKSMS/src/main/java/com/moez/QKSMS/common/utils/DateFormatter.java [hunk 2]
```diff
@@ -11,65 +11,90 @@
...
+        if (isSameYear(date)) {
+            return accountForFlippedDayMonth(accountFor24HourTime(context, new SimpleDateFormat("MMM d, h:mm a"))).format(date);
         }
 
-        return new SimpleDateFormat(isUsing24HourTime ? "MMM d y, H:mm" : "MMM d y, h:mm a").format(date);
+        return accountForFlippedDayMonth(accountFor24HourTime(context, new SimpleDateFormat("MMM d y, h:mm a"))).format(date);
     }
 
     public static String getDate(Context context, long date) {
         SharedPreferences prefs = MainActivity.getPrefs(context);
```
Comment (moezbhatti) [inline QKSMS/src/main/java/com/moez/QKSMS/common/utils/DateFormatter.java]: Unused variable

**Comment by moezbhatti** [comment]
Looks good to me
Just clean up those unused variables and we can merge

**Comment by moezbhatti** [comment]
Can you rebase off of develop?

**Comment by l3d00m** [comment]
"Just now" (line102) is Hard-coded again. Can you change the line to its original?

**Pull request closed without merging**